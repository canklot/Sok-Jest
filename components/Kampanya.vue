<template>
  <!-- Kampanyalardan 9 tane falan olcak v-for ile döngü şeklinde store'dan çekecek -->
  <div class="kampanya">
    <img src="../images//kampanya1.jpg" />
    <p class="metin">{{Parametre}}</p>
    <p class="kampanyadetay">Kampanya Detayları ></p>
  </div>
</template>

<script>
export default {
  props: {
    Parametre: {
      type: String,
      default: "Ben defaultum "
    }
  }
};
</script>

<style scoped>
.kampanya {
  margin-right: 30px;
  height: 240px;
  width: 364px;
  float: left;
  margin-top: 30px;
}
.kampanya img {
  width: 363px;
  border-radius: 3px;
  cursor: pointer;
}
.metin {
  width: 363px;
  font-family: Arial, Helvetica, sans-serif;
  margin-top: 10px;
  font-size: 15px;
  height: 80px;
}
.kampanyadetay {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 600;
  margin-top: 10px;
  cursor: pointer;
}
</style>