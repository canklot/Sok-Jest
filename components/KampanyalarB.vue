<template>
  <div>
    <div class="header">
      <div class="kampanyalar">
        <!-- Kampanyalardan 9 tane falan olcak v-for ile döngü şeklinde store'dan çekecek -->
        <Kampanya :Parametre="salad" />
        
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from "vuex";
import store from "../store/store.js";
import Vue from "vue";
import Vuex from "vuex";
import Kampanya from "./Kampanya.vue";
Vue.use(Vuex);

export default {
  data() {
    return {
      ...this.$store.state
        .store /* sonuna neden tekrar store koyuyuorum anlamadım ama silince çalışmıyor*/
    };
  },
  methods: {
    ...mapActions(["addIngredient"])
  },
  components: {
    Kampanya
  }
};
</script>

<style scoped>
.kampanyalar {
  width: 1200px;
  margin-left: 250px;
  margin-top: 30px;
}
.kampanya {
  margin-right: 30px;
  height: 240px;
  width: 364px;
  float: left;
  margin-top: 30px;
}
.kampanya img {
  width: 363px;
  border-radius: 3px;
  cursor: pointer;
}
.metin {
  width: 363px;
  font-family: Arial, Helvetica, sans-serif;
  margin-top: 10px;
  font-size: 15px;
  height: 80px;
}
.kampanyadetay {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 600;
  margin-top: 10px;
  cursor: pointer;
}
</style>