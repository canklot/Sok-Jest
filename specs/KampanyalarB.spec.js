import Vuex from "vuex";
import Vue from "vue";
import { mount, createLocalVue } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";

import KampanyalarB from "../components/KampanyalarB.vue";
import Kampanya from "../components/Kampanya.vue";
import saladStore from "../store/store.js";

/* const VueWithVuex = createLocalVue()
VueWithVuex.use(Vuex) */

Vue.use(Vuex);

test("store is loaded ", () => {
  const store = new Vuex.Store(saladStore);
  const wrapper = mount(KampanyalarB, { store });

  /* store.state.salad.push('cucumber') */
  /* console.log("looooooooooooooooooooog");
  console.log(store.state.salad); */

  /* Store dan vuex ile çakilen veriyi kontrol ediyor */
  expect(store.state.salad).toEqual([
    "25 Tl ve üzeri alışverişinize SuperFresh Pizza King 780 gr sadece 12 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize otomatik yansır.",
    "25 Tl ve üzeri alışverişinize Lipton Altın Harman Demlik Poşet 100 lü sadece 14 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize",
    "Nescafe Gold 150 gr Nestle Çikolata 60 gr sadece 32,45 TL İkili özel paket!",
    "25 Tl ve üzeri alışverişinize SuperFresh Pizza King 780 gr sadece 12 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize otomatik yansır. ",
    "25 Tl ve üzeri alışverişinize Ülker Çokokrem Cam Kavanoz 650 gr sadece 14 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize otomatik",
    "25 Tl ve üzeri alışverişinize Loreal Kolajen Bakımı Günlük Bakım Kremi 50m sadece 25 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize",
    "25 Tl ve üzeri alışverişinize Loreal Kolajen Bakımı Günlük Bakım Kremi 50m sadece 25 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize",
    "25 Tl ve üzeri alışverişinize Activex Antibakteriyel Islak Havlu 2x56 Adet sadece 11 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize",
    "25 Tl ve üzeri alışverişinize Heinz Ketçap 700 Gr Mayonez 400 Gr sadece 19 TL ! Sepetiniz 25tl ve üzeri ise ürünü sepete eklediğinizde indirimli fiyat sepetinize otomatik"
  ]);
});

describe("Foo", () => {
  
    const store = new Vuex.Store(saladStore);
    const wrapper = mount(KampanyalarB, { store });
    /* console.log("looooooooooooooooooooog");
    console.log(wrapper.html()); */
    it("has the expected html structure", () => {
      expect(wrapper).toMatchSnapshot();
    });
  
});

/* test("component renders the users", () => {
  const store = new Vuex.Store(saladStore);
  const wrapper = shallowMount(KampanyalarB, { store },
    {slots: {
    Kampanya: Kampanya
  }},);
  console.log("looooooooooooooooooooog");
  console.log(wrapper.html());
  const lis = wrapper.findAll("Kampanya"); 
  expect(lis.length).toBe(9);
}

);
 */
