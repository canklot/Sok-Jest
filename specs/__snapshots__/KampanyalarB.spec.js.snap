// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storeile Ã§ekilen vuexin renderini test et has the expected html structure 1`] = `
<div>
  <div class="header">
    <div class="kampanyalar"></div>
  </div>
</div>
`;
